format/negentropy/negentropy.go

```diff
@@ -0,0 +1,114 @@
+package negentropy
+
+// https://github.com/hoytech/negentropy
+
+import (
+	"embed"
+	"math"
+	"time"
+
+	"github.com/wader/fq/format"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/interp"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+//go:embed negentropy.md
+var negFS embed.FS
+
+func init() {
+	interp.RegisterFormat(
+		format.Negentropy,
+		&decode.Format{
+			Description: "Negentropy message",
+			DecodeFn:    decodeNegentropyMessage,
+			Groups:      []*decode.Group{},
+		})
+	interp.RegisterFS(negFS)
+}
+
+const (
+	version         = 0x61
+	fingerprintSize = 16
+)
+
+var modeMapper = scalar.SintMapSymStr{
+	0: "skip",
+	1: "fingerprint",
+	2: "idlist",
+}
+
+type timestampTranslator struct{ last time.Time }
+
+func (tt *timestampTranslator) MapSint(s scalar.Sint) (scalar.Sint, error) {
+	if s.Actual == 0 {
+		s.Description = "infinity"
+		tt.last = time.Unix(math.MaxInt64, 0)
+		return s, nil
+	} else {
+		timestamp := tt.last.Add(time.Second * time.Duration(s.Actual-1))
+		s.Description = timestamp.UTC().Format(time.DateTime + " UTC")
+		s.Sym = timestamp.Unix()
+		tt.last = timestamp
+		return s, nil
+	}
+}
+
+func decodeNegentropyMessage(d *decode.D) any {
+	tt := &timestampTranslator{last: time.Unix(0, 0)}
+
+	d.Endian = decode.BigEndian
+
+	v := d.FieldU8("version")
+	if v != version {
+		d.Fatalf("unexpected version %d (expected %d), is this really a negentropy message?", v, version)
+	}
+
+	d.FieldStructArrayLoop("bounds", "bound", d.NotEnd, func(d *decode.D) {
+		d.FieldSintFn("timestamp", decodeVarInt, tt)
```

Comment:
If you want to also have delta you could use a "synthetics" field and decode it something like this:
```
delta := d.FieldSintFn("timestamp_delta", decodeVarInt)
d.FieldSintValue("timestamp", delta, tt)
```
but as actual is the delta and sym is abs now maybe it's not needed? just a note that it's possible to do such things

Reply:
Cool, this is what I wanted since the beginning! I saw there was some "synthetic" stuff in the API but couldn't figure out how to use it (nor I expected it to be what I wanted). Thanks.